syntax = "proto3";

message CreateRequest {
  string handle = 1;
  string password = 2;
  string display_name = 3;
  string bio = 4;
}

message CreateResponse {
  enum ResultType {
    OK    = 0;
    ERROR = 1;
  }

  ResultType result_type = 1;
  string error_details = 2;
}

message LoginRequest {
  string handle = 1;
  string password = 2;
}

message LoginResponse {
  enum ResultType {
    ACCEPTED = 0;
    DENIED   = 1;
    ERROR    = 2;
  }

  ResultType result = 1;
  string error_details = 2;
  // The token the user should send with future requests.
  string token = 3;
}

message Token {
  string token = 1;
}

message TokenResponse {
  enum ResultType {
    OK    = 0;
    ERROR = 1;
  }

  ResultType result = 1;
  string error_details = 2;

  // Set if result is OK
  int64 global_id = 3;
  int64 handle = 4;
}

service Users {
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc Create(CreateRequest) returns (CreateResponse);
  rpc CheckToken(Token) returns (TokenResponse);
}
